WcaOnRails/app/models/competition.rb

```diff
@@ -394,7 +399,7 @@ def end_date
   end
 
   private def dates_must_be_valid
-    if !isConfirmed && self.year == 0 && self.month == 0 && self.day == 0 && @endYear == 0 && self.endMonth == 0 && self.endDay == 0
+    if !isConfirmed && !showAtAll && self.year == 0 && self.month == 0 && self.day == 0 && @endYear == 0 && self.endMonth == 0 && self.endDay == 0
```

Comment:
Since you added a new method let's make use of it
`!isConfirmed && !showAtAll` -> '!confirmed_or_visible?`
Furthermore I think we should refacor the rest too, like so:
```ruby
if !confirmed_or_visible? && [year, month, day, @endYear, endMonth, endDay].all? { |n| n == 0 }
```